week02/Group Anagrams.cpp

```diff
@@ -0,0 +1,30 @@
+// TC : O(NK)  N is the size of strs and K is the max size of a string of strs
+// SC : O(NK)
+class Solution {
+public:
+    
+    unsigned long long computeHash(string a){
+        unsigned long long result=1;    
+        for(char c  : a){
+            
+            result*=(257)*(c);
+        }
+        return result;
+    }
+    vector<vector<string>> groupAnagrams(vector<string>& strs) {
+
+        unordered_map<unsigned long long, vector<string> > bucket;
+        vector< vector<string> > result;
+        for(int i=0;i< strs.size();i++)
+        {
+            bucket.push_back(computeHash(strs[i]), strs[i]);
+        }    
+
+        for(auto it : bucket){
+
+            vector<int> temp= it->second;
+            result.push_back(temp);
+        }
+        return result;
+    }
```

Comment:
```suggestion
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<unsigned long long, vector<string> > bucket;
        vector< vector<string> > result;
        for(auto &str : strs) {
            bucket[computeHash(str)].push_back(str);
        }    
        for(auto group : bucket){
            result.push_back(group.second);
        }
        return result;
    }
```

Does your code compile? And you need to work on your variable namings and coding style.